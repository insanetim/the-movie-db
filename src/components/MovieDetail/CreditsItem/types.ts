export interface CreditsItemProps {
  profilePath: string | null
  title: string
  description: string
}
