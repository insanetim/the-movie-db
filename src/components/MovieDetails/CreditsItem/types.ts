export type CreditsItemHookProps = {
  id: number
  title: string
}

export type CreditsItemHookReturn = {
  handleNavigateToPerson: () => void
}

export type CreditsItemProps = {
  description?: string
  id: number
  profilePath?: string
  title: string
}
