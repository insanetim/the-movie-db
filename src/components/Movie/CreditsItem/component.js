import PropTypes from 'prop-types'
import { Col, Card } from 'antd'

import NoImage from 'src/assets/images/no-image.svg'

const CreditsItem = ({ profilePath, title, description }) => (
  <Col
    xs={{ span: 24 }}
    sm={{ span: 12 }}
    md={{ span: 8 }}
    lg={{ span: 8 }}
    xl={{ span: 6 }}
  >
    <Card
      className='top-margin'
      cover={
        profilePath ? (
          <img
            src={`https://image.tmdb.org/t/p/w500${profilePath}`}
            alt='Profile'
          />
        ) : (
          <div className='ant-card-cover--no-image'>
            <img
              src={NoImage}
              alt=''
            />
          </div>
        )
      }
    >
      <Card.Meta
        title={title}
        description={description}
      />
    </Card>
  </Col>
)

CreditsItem.propTypes = {
  profilePath: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string
}

CreditsItem.defaultProps = {
  profilePath: '',
  title: '',
  description: ''
}

export default CreditsItem
