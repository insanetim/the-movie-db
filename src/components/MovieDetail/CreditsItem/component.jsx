import PropTypes from 'prop-types'
import { Card } from 'antd'

import NoImage from 'src/assets/images/no-image.svg'

const CreditsItem = ({ profilePath, title, description }) => (
  <Card
    cover={
      profilePath ? (
        <img
          src={`https://image.tmdb.org/t/p/w500${profilePath}`}
          alt='Profile'
        />
      ) : (
        <div className='ant-card-cover--no-image'>
          <img
            src={NoImage}
            alt=''
          />
        </div>
      )
    }
  >
    <Card.Meta
      title={title}
      description={description}
    />
  </Card>
)

CreditsItem.propTypes = {
  profilePath: PropTypes.string,
  title: PropTypes.string,
  description: PropTypes.string
}

CreditsItem.defaultProps = {
  profilePath: '',
  title: '',
  description: ''
}

export default CreditsItem
