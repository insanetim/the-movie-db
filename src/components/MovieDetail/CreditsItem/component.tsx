import { Card } from 'antd'
import NoImage from 'src/assets/images/no-image.svg'
import isNull from 'src/utils/helpers/isNull'

import type { CreditsItemProps } from './types'

const CreditsItem: React.FC<CreditsItemProps> = ({
  description,
  profilePath,
  title
}) => {
  let cover = (
    <div className='ant-card-cover--no-image'>
      <img
        alt={title}
        src={NoImage}
      />
    </div>
  )
  if (!isNull(profilePath)) {
    cover = (
      <img
        alt={title}
        src={`https://image.tmdb.org/t/p/w500${profilePath}`}
      />
    )
  }

  return (
    <Card cover={cover}>
      <Card.Meta
        description={description}
        title={title}
      />
    </Card>
  )
}

export default CreditsItem
