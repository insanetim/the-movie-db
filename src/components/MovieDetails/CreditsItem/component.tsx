import { Card } from 'antd'
import NoImage from 'src/assets/images/no-image.svg'

import useContainer from './hook'
import { CreditsItemProps } from './types'

const CreditsItem: React.FC<CreditsItemProps> = ({
  description,
  id,
  profilePath,
  title,
}) => {
  const { handleNavigateToPerson } = useContainer({ id, title })

  let cover: JSX.Element
  if (profilePath) {
    cover = (
      <img
        alt={title}
        src={`https://image.tmdb.org/t/p/w500${profilePath}`}
      />
    )
  } else {
    cover = (
      <div className='ant-card-cover--no-image'>
        <img
          alt={title}
          src={NoImage}
        />
      </div>
    )
  }

  return (
    <Card
      cover={cover}
      hoverable
      onClick={handleNavigateToPerson}
    >
      <Card.Meta
        description={description}
        title={title}
      />
    </Card>
  )
}

export default CreditsItem
