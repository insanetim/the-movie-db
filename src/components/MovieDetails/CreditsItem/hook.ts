import { useNavigate } from 'react-router-dom'
import getSlug from 'src/utils/helpers/getSlug'

import { CreditsItemHookProps, CreditsItemHookReturn } from './types'

const useContainer = ({
  id,
  title,
}: CreditsItemHookProps): CreditsItemHookReturn => {
  const navigate = useNavigate()

  const handleNavigateToPerson = () => {
    navigate(`/person/${getSlug(id, title)}`)
  }

  return { handleNavigateToPerson }
}

export default useContainer
