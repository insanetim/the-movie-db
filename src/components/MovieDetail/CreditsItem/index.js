import CreditsItem from './component'

export default CreditsItem
