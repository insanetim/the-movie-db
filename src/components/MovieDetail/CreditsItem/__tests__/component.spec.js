import { render } from '@testing-library/react'

import CreditsItem from '../component'

describe('CreditsItem component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<CreditsItem profilePath='test/image' />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot with NoImage', () => {
    const { asFragment } = render(<CreditsItem profilePath='' />)

    expect(asFragment()).toMatchSnapshot()
  })
})
