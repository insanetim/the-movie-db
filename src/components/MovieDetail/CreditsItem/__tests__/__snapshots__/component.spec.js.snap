// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsItem component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1vtf12y"
  >
    <div
      class="ant-card-cover"
    >
      <img
        alt="Profile"
        src="https://image.tmdb.org/t/p/w500test/image"
      />
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`CreditsItem component matches snapshot with NoImage 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered css-dev-only-do-not-override-1vtf12y"
  >
    <div
      class="ant-card-cover"
    >
      <div
        class="ant-card-cover--no-image"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      />
    </div>
  </div>
</DocumentFragment>
`;
